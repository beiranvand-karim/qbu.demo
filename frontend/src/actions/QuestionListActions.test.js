/* global describe, it, expect */
import configureStore from 'redux-mock-store'
import {
   fetchQuestionListBegin,
   fetchQuestionListSuccess,
   fetchQuestionListError
} from './QuestionListActions'

describe('QuestionListActions', () => {

   const mockStore = configureStore();
   const store = mockStore();

   const questionList = [
      {
         "id": 7,
         "title": "this is title",
         "text": "this is description",
         "prize": 45000,
         "userId": 7,
         "username": "admin"
      }
   ];

   beforeEach(() => {
      store.clearActions();
   });

   it('should matches the question list begin snapshot', () => {
      store.dispatch(fetchQuestionListBegin());
      expect(store.getActions()).toMatchSnapshot();
   });

   it('should matches the question list success snapshot', () => {
      store.dispatch(fetchQuestionListSuccess(questionList));
      expect(store.getActions()).toMatchSnapshot();
   });

   it('should matches the question list error snapshot', () => {
      store.dispatch(fetchQuestionListError('this is a error'));
      expect(store.getActions()).toMatchSnapshot();
   });
});