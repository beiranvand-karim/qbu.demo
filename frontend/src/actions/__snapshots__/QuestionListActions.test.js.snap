// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionListActions should matches the question list begin snapshot 1`] = `
Array [
  Object {
    "type": "FETCH_QUESTION_LIST_BEGIN",
  },
]
`;

exports[`QuestionListActions should matches the question list error snapshot 1`] = `
Array [
  Object {
    "payload": "this is a error",
    "type": "FETCH_QUESTION_LIST_ERROR",
  },
]
`;

exports[`QuestionListActions should matches the question list success snapshot 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "id": 7,
        "prize": 45000,
        "text": "this is description",
        "title": "this is title",
        "userId": 7,
        "username": "admin",
      },
    ],
    "type": "FETCH_QUESTION_LIST_SUCCESS",
  },
]
`;
